cpp/src/Slice/Parser.cpp

```diff
@@ -356,24 +356,12 @@ Slice::SyntaxTreeBase::unit() const
     return _unit;
 }
 
-DefinitionContextPtr
-Slice::SyntaxTreeBase::definitionContext() const
-{
-    return _definitionContext;
-}
-
 void
 Slice::SyntaxTreeBase::visit(ParserVisitor* /*visitor*/)
 {
 }
 
-Slice::SyntaxTreeBase::SyntaxTreeBase(UnitPtr unit) : _unit(std::move(unit))
-{
-    if (_unit)
-    {
-        _definitionContext = _unit->currentDefinitionContext();
-    }
```

Comment:
Because of the round-about way we create new instances of Unit, this block is never entered.